src/lib/logical_query_plan/lqp_translator.cpp

```diff
@@ -208,15 +208,31 @@ std::shared_ptr<AbstractOperator> LQPTranslator::_translate_predicate_node_to_in
   if (value2_variant) right_values2.emplace_back(*value2_variant);
 
   const auto stored_table_node = std::dynamic_pointer_cast<StoredTableNode>(node->left_input());
+  const auto pruned_chunk_ids = stored_table_node->pruned_chunk_ids();
+
   const auto table_name = stored_table_node->table_name;
   const auto table = Hyrise::get().storage_manager.get_table(table_name);
   std::vector<ChunkID> indexed_chunks;
 
+  auto pruned_table_chunk_id = 0;
+  auto pruned_vector_id = 0;
```

Comment:
You could also use an iterator here and add `if (pruned_chunk_ids_iter == pruned_chunk_ids.end()) break` below